postgres-appliance/configure_spilo.py

```diff
@@ -210,16 +211,21 @@ def deep_update(a, b):
 
 def get_provider():
     try:
-        logging.info("Figuring out my environment (Google? AWS? Local?)")
+        logging.info("Figuring out my environment (Google? AWS? Openstack? Local?)")
         r = requests.get('http://169.254.169.254', timeout=2)
         if r.headers.get('Metadata-Flavor', '') == 'Google':
```

Comment:
I am wondering, may be on Openstack there is something similar available?'
Unique header for example.

Reply:
from what I can tell there is not.  I ran the following to check
```
root@patroni-patroni-0:/home/postgres# python3
Python 3.4.3 (default, Nov 17 2016, 01:08:31) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import requests
>>> r = requests.get('http://169.254.169.254', timeout=2)
>>> r.headers
{'Date': 'Thu, 07 Sep 2017 22:51:27 GMT', 'Content-Type': 'text/plain; charset=UTF-8', 'Connection': 'keep-alive', 'Content-Length': '98'}
```
I did find that there is an openstack specific path 
`http://169.254.169.254/openstack/latest/meta_data.json`
perhaps this would be a more reliable method of determining that a host is running on openstack.